mapserver/django/django_project/mapserver/api/wms.py

```diff
@@ -0,0 +1,79 @@
+__author__ = 'Irwan Fathurrahman <irwan@kartoza.com>'
+__date__ = '09/06/20'
+
+import json
+import os
+import requests
+from urllib.parse import urlparse, parse_qs, urlencode
+from django.conf import settings
+from django.http import (
+    HttpResponse, HttpResponseForbidden, HttpResponseBadRequest,
+    HttpResponseServerError)
+
+
+def get_sld_style(layer_name):
+    """ return sld style in json
+    :param layer_name: the layer name that needs to be checked
+    :return:
+    """
+    fixture_file = os.path.join(settings.FIXTURES, 'data', 'layer_style.json')
+    with open(fixture_file, 'r') as _file:
+        sld = json.load(_file).get(layer_name, None)
+        if sld:
+            return os.path.join(settings.MAPSERVER_PUBLIC_SLD_URL, sld)
+    return None
+
+
+def redirect_mapserver(request, mapserver_url):
+    """ Return updated url for redirect to mapserver
+    :param request:
+    :return: params
+    :rtype: str
+    """
+    try:
+        if request.method == 'GET':
+            # Parse querystring
+            parse_result = urlparse(request.build_absolute_uri())
+            query_params = parse_qs(parse_result.query)
```

Comment:
@meomancer please use this from now on when we are modifying a query params.

Reply:
i used the "non parsing params" because this API will be used a lot
and i didn't want parse anymore to prevent any loop in API